## Дз № 2

### Условие 1: 1 задача
Скачать файл в уроке 'kc_house_data.csv'
Считать данные с помощью pandas
Вывести на экран первые 5 строк
Посмотреть на описание признаков и на их содержание


In [2]:
import pandas as pd

# data = './kc_house_data.csv'
data = pd.read_csv('./kc_house_data.csv')
# data
data.head(5)

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

### Условие 2: 2 задача
Проведите первичный анализ данных. <br>Изучите типы данных. <br>Найдите количество пропущенных ячеек в данных. <br>Посчитайте основные статистики по всем признакам и поизучайте их. <br>Пишите выводы

In [3]:
data.info()
data.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

id               0
date             0
price            0
bedrooms         0
bathrooms        0
sqft_living      0
sqft_lot         0
floors           0
waterfront       0
view             0
condition        0
grade            0
sqft_above       0
sqft_basement    0
yr_built         0
yr_renovated     0
zipcode          0
lat              0
long             0
sqft_living15    0
sqft_lot15       0
dtype: int64

Данные состоят из 20-ти элементов, принимающих типы данных float - 5, int64 - 15, object - 1. <br> Данные структурированы и выводятся в 21 колонку. Количество пропущенных ячеек - 0. <br>Общее количество элементов - 21613, размер файла 3,5 мб.

### Условие 3: 3 задача
Ответьте на несколько вопросов
<br>3.1 В каком диапазоне изменяются стоимости недвижимости?

In [13]:
# pd.set_option('display.float_format', lambda x: '%.f' % x)
data.describe()
# data.shape
# data['price'].min(), data['price'].max()
# data.describe(include='object')
# data['sqft_living'].value_counts()
# data['sqft_living'].mode()


id   price  bedrooms  bathrooms  sqft_living  sqft_lot  floors  \
count      21613   21613     21613      21613        21613     21613   21613   
mean  4580301521  540088         3          2         2080     15107       1   
std   2876565571  367127         1          1          918     41421       1   
min      1000102   75000         0          0          290       520       1   
25%   2123049194  321950         3          2         1427      5040       1   
50%   3904930410  450000         3          2         1910      7618       2   
75%   7308900445  645000         4          2         2550     10688       2   
max   9900000190 7700000        33          8        13540   1651359       4   

       waterfront  view  condition  grade  sqft_above  sqft_basement  \
count       21613 21613      21613  21613       21613          21613   
mean            0     0          3      8        1788            292   
std             0     1          1      1         828            443   
min             0     0          1      1         290              0   
25%             0     0          3      7        1190              0   
50%             0     0          3      7        1560              0   
75%             0     0          4      8        2210            560   
max             1     4          5     13        9410           4820   

       yr_built  yr_renovated  zipcode   lat  long  sqft_living15  sqft_lot15  
count     21613         21613    21613 21613 21613          21613       21613  
mean       1971            84    98078    48  -122           1987       12768  
std          29           402       54     0     0            685       27304  
min        1900             0    98001    47  -123            399         651  
25%        1951             0    98033    47  -122           1490        5100  
50%        1975             0    98065    48  -122           1840        7620  
75%        1997             0    98118    48  -122           2360       10083  
max        2015          2015    98199    48  -121           6210      871200

Диапозон стоимости недвижимости от 75000 до 7700000

<br>3.2 Какую долю в среднем занимают жилая площадь от всей площади по всем домам?

In [30]:
all_sqft = data['sqft_lot'].mean()
liv_sqft = data['sqft_living'].mean()
ave_liv_sqft = (liv_sqft / all_sqft) * 100
ave_liv_sqft

13.76781757959227

<br>3.3 Как много домов с разными этажами в данных?


In [31]:
data['floors'].value_counts()

1    10680
2     8241
2     1910
3      613
2      161
4        8
Name: floors, dtype: int64

<br>3.4 Насколько хорошие состояния у домов в данных?

In [32]:
data['condition'].value_counts()

3    14031
4     5679
5     1701
2      172
1       30
Name: condition, dtype: int64

<br>3.5 Найдите года, когда построили первый дом, когда построили последний дом в данных?

In [33]:
data['yr_built'].min(), data['yr_built'].max()

(1900, 2015)

### Условие 4: 
Ответьте на несколько вопросов
<br>4.1 Сколько в среднем стоят дома, у которых 2 спальни?

In [9]:
data[data['bedrooms'] == 2]['price'].mean()

401372.681884058

4.2 Какая в среднем общая площадь домов, у которых стоимость больше 600 000?

In [10]:
data[data['price'] > 600000]['sqft_lot'].mean()

20442.524776214832

4.3 Как много домов коснулся ремонт?

In [13]:
data[(data['yr_renovated'] != 0)].shape[0]

914

4.4 Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?

In [17]:
grade10_mean_price = data[data['grade'] > 10]['price'].mean()
grade4_mean_price = data[data['grade'] < 4]['price'].mean()

round(grade10_mean_price - grade4_mean_price, 2)

1488885.12

### Условие 5:<br>Ответьте на несколько вопросов
<br>5.1 Выберите дом клиенту. Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента?

In [18]:
data[
    (data['waterfront'] == 1)
    & (data['bathrooms'] > 3)
    & (data['sqft_basement'] != 0)
].shape[0]

36

5.2 Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома?

In [20]:
client_want = (
    (data['view'] == 4)
    | (data['waterfront'] == 1)
    & (data['condition'] == 5)
    & (data['yr_built'] >= 1980)
)

res = data[client_want]
res['price'].min(), res['price'].max()

(252000.0, 6885000.0)

5.3 Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?

In [21]:
client_req = data[
    (data['sqft_basement'] == 0)
    & (data['floors'] == 2)
    & (data['price'] < 150000)
]
client_req['condition'].mean()

2.8333333333333335